### ISSUE TITLE
Add Keepalive Option to Miner

### ISSUE THREAD

**Issue by setuidroot** [todxx/teamredminer#83]
I've been using TRM for awhile now mining Monero on MoneroOcean (nodejs) pool... everything has worked fine with TRM getting good hashrates.  That is until last night when TRM went 15 minutes without submitting any shares to the pool and the pool unauthenticated TRM miner due to it timing out.  Well TRM kept mining all night long (~12 hours using lots of power mining) sending in shares to the pool after the connection had been unauthenticated by the pool.  

This resulted in 12+ hours of completely lost work as the pool had closed the connection and quit paying attention to the shares TRM was sending.  TRM showed ~1467 rejected shares due to being "unauthenticated" with the pool; the pool unfortunately didn't even register these rejected shares, so it's completely wasted work by my GPUs :(

Please add a keepalive function, or if you have one already, let's add some documentation about it.  I think keepalive should be configured by default to avoid people ending up in the situation I'm in now.

I'm using an older TRM version (0.44 or 0.45) I haven't bothered to update it because it's been running very well up until last night.  Maybe you have keepalive in newer versions, but I think it should be the default behavior of the miner for pools that accept it.

**Comment by todxx** [comment]
Hi, sorry for not getting back to you on this issue earlier.  The problem you're running into is a pool-side issue.  The connection is never closed by the pool, it simply decides at some point that it will reject all future submitted shares without notifying the miner in any way.  

To work around pools with this problem, in version 0.5.8 we added a check for consecutive rejected shares.  Now by default the miner will reconnect to the pool if it sees more than 5 consecutive rejected shares.  The limit can also be adjusted by command line options.

As a side note, I suggest using the lastest v0.5.9 as v0.5.8 ended up hitting some linux driver issues and can crash on some machines. In v0.5.9 we implemented workarounds for the driver issues.
